engines/startrek/detection.cpp

```diff
@@ -0,0 +1,394 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+#include "base/plugins.h"
+
+#include "engines/advancedDetector.h"
+
+#include "graphics/thumbnail.h"
+
+#include "common/config-manager.h"
+#include "common/file.h"
+#include "common/savefile.h"
+
+#include "startrek/startrek.h"
+
+namespace StarTrek {
+
+struct StarTrekGameDescription {
+	ADGameDescription desc;
+
+	uint8 gameType;
+	uint32 features;
+};
+
+uint32 StarTrekEngine::getFeatures() const {
+	return _gameDescription->features;
+}
+
+Common::Platform StarTrekEngine::getPlatform() const {
+	return _gameDescription->desc.platform;
+}
+
+uint8 StarTrekEngine::getGameType() const {
+	return _gameDescription->gameType;
+}
+
+Common::Language StarTrekEngine::getLanguage() const {
+	return _gameDescription->desc.language;
+}
+
+} // End of Namespace StarTrek
+
+static const PlainGameDescriptor starTrekGames[] = {
+	{"startrek", "Star Trek game"},
+	{"st25", "Star Trek: 25th Anniversary"},
+	{"stjr", "Star Trek: Judgment Rites"},
+	{0, 0}
+};
+
+
+namespace StarTrek {
+
+static const StarTrekGameDescription gameDescriptions[] = {
+	// TODO: Replace AD_ENTRY1 with AD_ENTRY1s for game versions I don't have yet
+
+	{ // ST25 DOS CD-ROM edition (EN)
+		{
+			"st25",
+			"CD",
+			AD_ENTRY1s("data.001", "57040928a0f374281aa86ba4e7db8444", 7793814),
+			Common::EN_ANY,
+			Common::kPlatformDOS,
+			ADGF_NO_FLAGS,
+			GUIO0()
+		},
+		GType_ST25,
+		GF_CDROM,
+	},
+
+	{ // ST25 DOS floppy edition (EN)
+		{
+			"st25",
+			"Floppy",
+			AD_ENTRY1s("data.001", "57040928a0f374281aa86ba4e7db8444", 7222630),
+			Common::EN_ANY,
+			Common::kPlatformDOS,
+			ADGF_NO_FLAGS,
+			GUIO0()
+		},
+		GType_ST25,
+		0,
+	},
+
+	{ // ST25 DOS floppy edition (GER)
+		{
+			"st25",
+			"Floppy",
+			AD_ENTRY1s("data.001", "57040928a0f374281aa86ba4e7db8444", 7181890),
+			Common::DE_DEU,
+			Common::kPlatformDOS,
+			ADGF_NO_FLAGS,
+			GUIO0()
+		},
+		GType_ST25,
+		0,
+	},
+
+	{ // ST25 Amiga ? (EN)
+		{
+			"st25",
+			"Floppy",
+			AD_ENTRY1("data.000", "f0918b6d096455ce2ae6dd5ef973292e"),
```

Comment:
Please, comment these out, so we could collect the filesizes. Users will be able to report them.